nautobot/extras/tests/test_relationships.py

```diff
@@ -103,8 +105,38 @@ def test_clean_filter_not_valid(self):
             type=RelationshipTypeChoices.TYPE_MANY_TO_MANY,
         )
 
-        with self.assertRaises(ValidationError):
+        with self.assertRaises(ValidationError) as handler:
             m2m.clean()
+        expected_errors = {"source_filter": ["'notvalid' is not a valid filter parameter for dcim.Site object"]}
+        self.assertEqual(handler.exception.message_dict, expected_errors)
+
+        m2m = Relationship(
+            name="Another Vlan to Rack",
+            slug="vlan-rack-2",
+            source_type=self.site_ct,
+            source_filter={"site": "not a list"},
+            destination_type=self.rack_ct,
+            type=RelationshipTypeChoices.TYPE_MANY_TO_MANY,
+        )
+
+        with self.assertRaises(ValidationError) as handler:
+            m2m.clean()
+        expected_errors = {"source_filter": ["'site' is not a valid filter parameter for dcim.Site object"]}
```

Comment:
Huh. That's not what I would have expected here - do we need to replace `"site"` with a different filter key/parameter? I would have thought that this test were expecting to see an error because `"not a list"` is not a list, not an error about the `"site"` key.